minecraft_go/software/mc_go/src/touchscreen.c

```diff
@@ -87,5 +67,33 @@ Point touchscreen_get_release(void) {
 	p2.y = (int) (buffer[4]) << 7;
 	p2.y += (int) (buffer[3]);
 
+	p2 = touchscreen_pixel_conversion(p2);
+
 	return p2;
 }
+
+void touchscreen_get_report_packet(unsigned char *buffer, unsigned char touchStatus){
+	/**
+	 * wait for a pen up command then return the X,Y coord of the point
+	 * calibrated correctly so that it maps to a pixel on screen
+	*/
+	while (buffer[0] != touchStatus) {
+		buffer[0] = serial_get_char(TOUCHSCREEN);
+	};
+
+	serial_get_n_char(TOUCHSCREEN, buffer + 1, sizeof(buffer));

```

Comment:
`sizeof(buffer)` is miraculously equal to 4 here. You'll have to pass the length of the buffer in as a parameter